overlord/configstate/tasksets.go

```diff
@@ -34,13 +36,25 @@ func init() {
 	snapstate.Configure = Configure
 }
 
+func maxConfigureHookRuntime() time.Duration {
+	maxRuntime := 5 * time.Minute
+	if mrs := os.Getenv("SNAPD_CONFIGURE_HOOK_TIMEOUT"); mrs != "" {
+		if mr, err := time.ParseDuration(mrs); err == nil {
+			maxRuntime = mr
+		}
+	}
+	return maxRuntime
+}
+
 // Configure returns a taskset to apply the given configuration patch.
 func Configure(s *state.State, snapName string, patch map[string]interface{}, flags int) *state.TaskSet {
 	hooksup := &hookstate.HookSetup{
 		Snap:       snapName,
 		Hook:       "configure",
 		Optional:   len(patch) == 0,
 		IgnoreFail: flags&snapstate.IgnoreHookFailure != 0,
+		// all configure hooks must finish within this timeout
+		MaxRuntime: maxConfigureHookRuntime(),
```

Comment:
Can we please standardize all language around "timeout", per the environment variable?  It sounds like this is the typical meaning of it, and HookSetup.Timeout and "timeout" feels pretty good vs. "max-runtime".